server/queues/tasks/CacheIssueSourcesTask.ts

```diff
@@ -0,0 +1,35 @@
+import { Integration } from "@server/models";
+import { sequelize } from "@server/storage/database";
+import { Hook, PluginManager } from "@server/utils/PluginManager";
+import BaseTask from "./BaseTask";
+
+const plugins = PluginManager.getHooks(Hook.IssueProvider);
+
+type Props = {
+  integrationId: string;
+};
+
+export default class CacheIssueSourcesTask extends BaseTask<Props> {
+  async perform({ integrationId }: Props) {
+    await sequelize.transaction(async (transaction) => {
+      const integration = await Integration.findByPk(integrationId, {
+        transaction,
+        lock: transaction.LOCK.UPDATE,
```

Comment:
Lets not take a lock the entire time `fetchSources` is running as it may take a very long time